### ISSUE TITLE
Several issues

### ISSUE THREAD

**Issue by happyDemon** [skiplagged/pokemongo-php#8]
I've been testing this package and the api for a little while now and I'm bumping into a few errors here and there.

**It loads pokemon outside of the bounding box I send through.**
This would be no problem for me if it also loaded pokemon from within the bounds.
I know that sometimes the data is not available, but when I do scan for pokemon (use the findPokemon method) there are some other issues.

**Using findPokemon yields inconsistent results**
I thought that if I used this method I would find the pokemon within my bounding box, as well as contribute the data to everyone else, but it doesn't appear to be too trustworthy.

Most of the time my server errors out (500).
Other times it throws exception (about missing response data).
once in a while I do get some data through though.

These 3 different scenarios arise randomly, while always using the same credentials.
Is it because your API server is hicking up sometimes?
are there ways this package can be stabilized?

Might local protobuf unpacking be an option to speed up the process of getting the data locally?

**Comment by marvantes** [comment]
Same here the connecting is unstable i guess doe the bussy servers from niantic but that protobuff unpaking on skipplaged server is too Long... I managed to limit the Pokémon feed only to max 50 responses :) and it worked but it tooks too much time :( anyway it is a little solution to do it in small areas and limit it... 

But still i Think that protobuff decompiling on own servers would do the job... I Hope they Will solve it for us soon..  i have a buissiness ideas and im willing to pay or donate something to autor for it...

**Comment by marvantes** [comment]
Are you open to help mě get node app working ? I need it seriously..

**Comment by aktarer** [comment]
> It loads pokemon outside of the bounding box I send through.

That's intentional. A box is computed to the nearest 2 degrees that contains your bounding box. This is for caching purposes as many users are panning within a a degree so it doesn't make sense to constantly run geospatial queries in the backend. You can easily filter out client side.

**Comment by aktarer** [comment]
> Using findPokemon yields inconsistent results

That live fetches spawned pokemon using your token. Should be pretty accurate.

**Comment by aktarer** [comment]
> Is it because your API server is hicking up sometimes?

Niantic's PTC login has been performing poorly today: http://ispokemongodownornot.com/

**Comment by happyDemon** [comment]
thank you for the answers, however that leaves the server error 500 unexplained.
By inconsistent results I meant the scenarios that followed.


Could I also add that when I do scan for pokemon and I get results back, when I refresh my page and query the same bounding box I scanned pokemon for, they are not in the results.

**Comment by aktarer** [comment]
>  however that leaves the server error 500 unexplained.

Our servers are being really hammered today. That api endpoint, `/api/pokemon.php` is being accessed by around 1,500 users simultaneously.. That's mostly due to the launch of our Android app: https://play.google.com/store/apps/details?id=com.skiplagged.pokemap We added servers and things seem to be working smoothly right now though. 

> Could I also add that when I do scan for pokemon and I get results back, when I refresh my page and query the same bounding box I scanned pokemon for, they are not in the results.

It's currently cached for 10 seconds. After 10 seconds, the results in the bounding box should automatically update.

**Comment by MichalCerny** [comment]
PTC login not working for me past 8 hours .. idk where is problem .. tried proxy/other internet providers and PTC is still in maintenance for me.

@marvantes napiš mi na fb /cerny.me .. dáme nějak hlavy dohromady ;)

**Comment by marvantes** [comment]
@aktarer why cant you give us that pokemon.php scritp? so we can use pokemon-api on our servers and not waiting for your?

**Comment by baskettcase** [comment]
@aktarer @marvantes I don't mind it hitting your server, I rather like the crowd sourced map, but it would be nice to have a variable that turns it on and turns it off if your server is struggling to keep up with the load.. either a variable we set, or if there is a timeout on the curl request and nothing is returned that it skips that part.. or both.

**Comment by wearelogik** [comment]
@aktarer Skiplagged is more unstable then Pokemon Go, and most of the errors come from Skiplagged, not Go.

**Comment by aktarer** [comment]
> Skiplagged is more unstable then Pokemon Go

We're currently handling over 30,000 requests per second to `/api/pokemon.php` in under a couple hundred milliseconds average response time.

> why cant you give us that pokemon.php scritp?

It's a virtual route and the file doesn't actually exist. The code supporting everything is in other languages.

> it's clear as day that the Skiplagged API has the most issues.

Requests are successfully handled 99%+ of the time.

> On top of that, it's a one-man show, just makes it worse.

It's not a one-man show.

**Comment by wearelogik** [comment]
When i first do get it to work, it's insanely slow.

Right now, running your example code as is, takes everything from 10-45 minutes to complete. Doesn't sound right?

**Comment by aktarer** [comment]
That sounds pretty excessive. We have this lib running on a box non-stop to monitor health. Calls to `skiplagged.com` seem fine. I recommend putting echo statements with times to figure out where the lag is. In the meantime, we'll audit our backend to see if something really is broken on our end.

**Comment by wearelogik** [comment]
It's not excessive when the code is running in an infinite loop and the loop provides a bunch of errors and then successfully call some pokemon every X minutes.

I am constantly given errors like

"failed to get login_url"
"failed to retrieve specific api endpoint"
"failed to get requests" --> All the time
"failed to get pdata" --> All the time

From start until error occurs (Without the while loop) still takes everything from 30-60 seconds.

**Comment by baskettcase** [comment]
Yes same here, Im getting a lot of errors. Like right now I just ran it and this is what I get:

1470013212.492 failed to retrieve specific api endpoint from SKIPLAGGED
1470013213.659 failed to get pdata from SKIPLAGGED
1470013214.826 failed to get pdata from SKIPLAGGED
1470013216.005 failed to get requests from SKIPLAGGED

I tagged each of the errors so that I knew where they were coming from, and so far today it has been all SkipLagged errors. Im using a cron in the background since as @wearelogik said, it's extremely slow if I run it real time.
